src/packages/solid/components/Input/Input.tsx

```diff
@@ -28,77 +28,108 @@ export interface InputProps extends IntrinsicNodeProps {
   actualTitle?: string;
 
   /**
-   * eyebrow of input container.
+   * index of the current cursor positions
    */
-  eyebrow?: string;
+  position?: number;
 
   /**
-   * help text for the input container.
+   * signal passed in to represent what change we want to happe in the input
    */
-  helpText?: string;
+  keyEvent: Signal<string>;
 
   /**
-   * index of the current cursor positions
+   * signal passed in to represent the actual title within the input
    */
-  position?: number;
-
-  keySignal: Signal<string>;
-
-  onRight?: KeyHandler;
-
-  onLeft?: KeyHandler;
+  titleSignal: Signal<string>;
 
   style?: Partial<InputStyles>;
 
   tone?: Tone;
 }
 
 const Input: Component<InputProps> = props => {
+  const [position, setPosition] = createSignal(props.position ?? props.actualTitle?.length ?? 0);
+
+  const formatInputText = key => {
+    if (key === undefined || key === '') {
+      return;
+    }
+
+    const inputText = props.titleSignal[0]();
+    let positions = position();
+    let newTitle = '';
+    switch (key.toLowerCase()) {
+      case 'backspace':
+      case 'delete':
+        newTitle = positions > 0 ? inputText.slice(0, positions - 1) + inputText.slice(positions) : inputText;
+        positions--;
+        break;
+      case 'done':
+        break;
+      case 'space':
+        newTitle =
+          positions > 0
+            ? inputText.slice(0, positions - 1) + ' ' + inputText.slice(positions)
+            : ' ' + inputText;
+        positions++;
+        break;
+      case 'clear':
+        newTitle = '';
+        positions = 0;
+        break;
+      default:
+        newTitle =
+          positions > 0 ? inputText.slice(0, positions) + key + inputText.slice(positions) : key + inputText;
+        positions++;
+        break;
+    }
+
+    props.keyEvent[1]('');
+    props.titleSignal[1](newTitle);
```

Comment:
Yes - those can be destructured since we're calling the signals. What you couldn't destructure is if this were a store.. meaning props.key = `value` (that is a proxy with getter + setter in one)

And I agree destructure it for cleaner code